File: XOutput.Api/Websocket/Input/InputDeviceDetailsRequest.cs

using System.Collections.Generic;
using XOutput.Common.Input;

namespace XOutput.Websocket.Input
{
    public class InputDeviceDetailsRequest : MessageBase
    {
        public const string MessageType = "InputDeviceDetails";
        public string Id { get; set; }
        public string Name { get; set; }
        public string HardwareId { get; set; }
        public List<InputDeviceSource> Sources { get; set; }
        public List<InputDeviceTarget> Targets { get; set; }
        public string InputApi { get; set; }
        public InputDeviceDetailsRequest()
        {
            Type = MessageType;
        }
    }
}

File: @xoutput/api/src/websocket/input/InputDeviceDetailsRequest.ts
import { MessageBase } from '../MessageBase';
import { InputDeviceSource } from '../../common/input/InputDeviceSource';
import { InputDeviceTarget } from '../../common/input/InputDeviceTarget';
import { InputDeviceApi } from '../../common/input/InputDeviceApi';

export const InputDeviceDetailsRequestType = 'InputDeviceDetails';

export type InputDeviceDetailsRequest = MessageBase<typeof InputDeviceDetailsRequestType> & {
  id: string;
  name: string;
  hardwareId?: string;
  sources: InputDeviceSource[];
  targets: InputDeviceTarget[];
  inputApi: InputDeviceApi;
};
